src/clojure_lsp/handlers.clj

```diff
@@ -489,6 +483,33 @@
         (log/warn "Could not apply" refactoring "to form: " (z/string loc)))
       (log/warn "Could not find a form at this location. row" line "col" column "file" doc-id))))
 
+(defn ^:private execute-refactor [command args]
+  (let [[doc-id line col & args] (map interop/json->clj args)]
+    (when-let [result (refactor doc-id
+                                (inc (int line))
+                                (inc (int col))
+                                command
+                                args)]
+      (.get (.applyEdit (:client @db/db)
+                        (ApplyWorkspaceEditParams.
+                          (interop/conform-or-log ::interop/workspace-edit result)))))))
+
+(defn ^:private execute-server-info []
+  (let [db @db/db
+        info {:project-root (:project-root db)
+              :project-settings (crawler/find-raw-project-settings (:project-root db))
+              :client-settings (:client-settings db)
+              :port (:port db)}]
+    (interop/conform-or-log ::interop/server-info info)))
```

Comment:
The interop specs are used for conforming into lsp4j objects and less about "type" checking.

I think before conforming, this should result in a clojure map like 
`{:message-type :info :message <string(markdown?) of server info>}` 
and then
`(conform-or-log ::interop/show-message result)` would build a ShowMessage instance in lsp4j https://microsoft.github.io/language-server-protocol/specifications/specification-current/#window_showMessage